src/main/java/io/cdap/plugin/cdc/transform/CDCStreamingToRecordTransform.java

```diff
@@ -0,0 +1,213 @@
+    }
+  }
+
+  @Override
+  public void transform(StructuredRecord input, Emitter<StructuredRecord> emitter) throws Exception {
+
+    for (Schema.Field field : input.getSchema().getFields()) {
+      String name = field.getName();
+      //LOG.info("Local field name " + field.getName());
+      //LOG.info("Field type " + field.getSchema().toString());
+
+      if (input.get(name) != null && name.equals("dml")) {
+
+        //LOG.info("Preparing record");
+        //LOG.info(input.get(name).toString());
+        StructuredRecord sr = input.get(name);
+
+        try {
+          StructuredRecord out = getDMLRecord(sr);
+          emitter.emit(out);
+        } catch (Exception ex) {
+        }
+
+      }
+    }
+
+  }
+
+  private StructuredRecord getDMLRecord(StructuredRecord input) throws Exception {
+
+    Schema outputSchema = null;
+    Schema outSchema = null;
+
+    for (Schema.Field field : input.getSchema().getFields()) {
+      String name = field.getName();
+
+      if (name.equals("rows_schema")) {
+
+        outSchema = Schema.parseJson(input.get(name).toString());
+
+      }
+    }
+
+    List<Schema.Field> fields = new ArrayList<>(outSchema.getFields().size() + 1);
+    fields.addAll(outSchema.getFields());
+    fields.add(Schema.Field.of("CDC_OP_TYPE", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+    fields.add(Schema.Field.of("CHANGE_TRACKING_VERSION", Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+
+    outputSchema = Schema.recordOf(outSchema.getRecordName() + ".added", fields);
+
+    //LOG.info("OutputSchema of new record " + outputSchema.toString());
+
+    StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
+
+    try {
+
+      for (Schema.Field field : input.getSchema().getFields()) {
+
+        String name = field.getName();
+
+        if (name.equals("rows_values")) {
+          Gson gson = new Gson();
+          JsonReader reader = new JsonReader(new StringReader(input.get("rows_values").toString()));
+          reader.setLenient(true);
+          Map<String, Object> rowsValues = new HashMap<>();
+          rowsValues = gson.fromJson(reader, Map.class);
+
+
+          for (Map.Entry<String, Object> entry : rowsValues.entrySet()) {
+
+            String type = "string";
+
+            for (Schema.Field fieldtype : outSchema.getFields()) {
+              if (fieldtype.getName().equals(entry.getKey())) {
+
+                type = fieldtype.getSchema().toString().split("\"")[1];
+                if (type.equals("type")) {
+                  type = fieldtype.getSchema().toString().split("\"")[3];
+                  if (fieldtype.getSchema().toString().split("\"")[5].equals("logicalType")) {
+                    type = fieldtype.getSchema().toString().split("\"")[7];
+                  }
+                }
+              }
+            }
+
+            type = type.trim().toLowerCase();
+
+            if (entry.getValue() == null || entry.getValue().toString() == null) {
```

Comment:
There is a lot of logic that parses strings generated in ResultSetToDMLRecord back into their original objects. I don't understand why ResultSetToDMLRecord was changed in the first place, then none of this parsing would be needed.

Reply:
This is because the rows_values was parsed into DML record but it was not parsed correctly for all the values. Especially dates, thats why it was did in such way/